File: LeetCode/3386/3386.cpp
class Solution {
public:
    int buttonWithLongestTime(vector<vector<int>>& events) {
        pair<int,int> result {events[0][0], events[0][1]};
        pair<int,int> prev = result;
        if(events.size() ==1 ) return result.first;
        for(auto iter = events.begin()+1; iter != events.end(); ++iter)
        {
            int raz = (*iter)[1] - prev.second;
            if(raz > result.second) result = {(*iter)[0], raz};
            if(raz == result.second && (*iter)[0] < result.first) result = make_pair((*iter)[0], raz);
            prev ={(*iter)[0], (*iter)[1]};
        }
        return result.first;
    }
};

File: LeetCode/3386/3386.rs
impl Solution {
    pub fn button_with_longest_time(events: Vec<Vec<i32>>) -> i32 {
        let mut result: (i32,i32) = (events[0][0], events[0][1]);
        let mut prev = result;
        events[1..].iter().for_each(|arr| {
            let raz = (*arr)[1] - prev.1;
            if result.1 < raz {result = ((*arr)[0], raz);}
            else if result.1 == raz && result.0 > (*arr)[0] {result = ((*arr)[0], raz);}
            prev = ((*arr)[0], (*arr)[1]);
        });
        result.0
        
    }
}
